File: solutions/javascript/bob/1/bob.js
/* eslint-disable no-unused-vars */
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

const isSilence = message => /^\s*$/.test(message)
const isShouting = message => message.toUpperCase() === message && /[A-Z]/.test(message)
const isAsking = message => /\?$/.test(message.trim())

// message.replace(/\s+/g, '') === ''
// message[message.length - 1] === '?'

export const hey = (message) => {
  if (isSilence(message)) {
    return 'Fine. Be that way!'
  }

  if (isShouting(message)) {
    return isAsking(message)
      ? "Calm down, I know what I'm doing!"
      : 'Whoa, chill out!'
  }

  return isAsking(message)
    ? 'Sure.'
    : 'Whatever.'
}

File: solutions/elixir/bob/1/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      String.length(String.trim(input)) == 0 -> "Fine. Be that way!"
      String.ends_with?(input, "?") -> cond do
        input == String.upcase(input) && input != String.downcase(input) -> "Calm down, I know what I'm doing!"
        true -> "Sure."
      end
      input == String.upcase(input) && input != String.downcase(input) -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end
end
